modules/angular/projects/troca-angular/src/lib/search-input/search-input.component.ts

```diff
@@ -0,0 +1,94 @@
+import {Component, EventEmitter, Input, OnInit, Output} from '@angular/core';
+import { FormControl, FormGroup } from '@angular/forms';
+import {InputComponent, InputType} from "../input/input.component";
+
+export interface ItemResults {
+  title :string,
+  description? :string,
+  icon? :string,
+  link? :string
+}
+
+
+@Component({
+  selector: 'tf-search-input',
+  templateUrl: './search-input.component.html',
+  styleUrls: ['./search-input.component.scss']
+})
+
+export class SearchInputComponent implements OnInit {
+  @Input() value?: any;
+  @Input() placeholder?: any;
+  @Input() className?: string;
+  @Input() formControl?: FormControl;
+  @Input() formControlName?: string;
+  @Input() formGroup?: FormGroup;
+  @Input() results?: ItemResults[];
+
+  @Output() onSelect?: EventEmitter<ItemResults> = new EventEmitter<ItemResults>();
+  @Output() onChange?: EventEmitter<KeyboardEvent> = new EventEmitter<KeyboardEvent>();
+  @Output() onClear?: EventEmitter<any> = new EventEmitter<any>();
+
+  public showCloseButton : boolean = false;
+
+  static readonly SEARCH_MIN_CHARACTERS = 3;
+
+  public resultOnFocus : number = -1;
+
+  getInputCharacters (keyboardEvent: KeyboardEvent) {
+    if(keyboardEvent.key === 'ArrowDown'){
```

Comment:
Aca creo que convendría hacerlo con una especie de Chain Of Responsibility, para que sea mas prolijo extenderlo.
Tambien, lo llamaria por lo que hace en vez de arrowDown(). O sea, algo asi como previousElement y nextElement.